src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/step/forcemerge/AttemptCallForceMergeStep.kt

```diff
@@ -42,35 +51,61 @@ class AttemptCallForceMergeStep(
     override fun isIdempotent() = false
 
     @Suppress("TooGenericExceptionCaught")
-    override suspend fun execute() {
+    override suspend fun execute(): AttemptCallForceMergeStep {
         try {
-            val indexName = managedIndexMetaData.index
 
-            logger.info("Attempting to force merge on [$indexName]")
+            val startTime = Instant.now().toEpochMilli()
             val request = ForceMergeRequest(indexName).maxNumSegments(config.maxNumSegments)
-            val response: ForceMergeResponse = client.admin().indices().suspendUntil { forceMerge(request, it) }
+            var response: ForceMergeResponse? = null
+            var throwable: Throwable? = null
+            GlobalScope.launch(Dispatchers.IO + CoroutineName("ISM-ForceMerge-$indexName")) {
+                try {
+                    response = client.admin().indices().suspendUntil { forceMerge(request, it) }
+                    if (response?.status == RestStatus.OK) {
+                        logger.info(getSuccessMessage(indexName))
+                    } else {
+                        logger.warn(getFailedMessage(indexName))
+                    }
+                } catch (t: Throwable) {
+                    throwable = t
+                }
+            }
+
+            while (response == null && (Instant.now().toEpochMilli() - startTime) < FIVE_MINUTES_IN_MILLIS) {
+                delay(FIVE_SECONDS_IN_MILLIS)
+                throwable?.let { throw it }
+            }
 
-            // If response is OK then the force merge operation has started
-            if (response.status == RestStatus.OK) {
+            val shadowedResponse = response
+            if (shadowedResponse?.let { it.status == RestStatus.OK } != false) {
```

Comment:
if shadowedResponse is null here, we still update success msg first right? Then it won't override possible failure msg

Reply:
If shadowed response is null it means we hit the 5 minute delay and just assume it's OK to proceed. It now returns a slightly different message saying it successfully called the force merge API whereas if there is a shadowedResponse then it returns a completed message.